src/extension/feature-pregeneralized/src/test/java/org/geoserver/data/gen/PreGeneralizedDataStoreFactoryTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2022, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geoserver.data.gen;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.net.URL;
+import org.geoserver.platform.GeoServerExtensionsHelper;
+import org.geoserver.platform.GeoServerResourceLoader;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.test.GeoServerSystemTestSupport;
+import org.geotools.data.gen.info.GeneralizationInfos;
+import org.geotools.util.URLs;
+import org.junit.Test;
+
+public class PreGeneralizedDataStoreFactoryTest extends GeoServerSystemTestSupport {
+    boolean configurationFileRequested = false;
```

Comment:
I would consider an instance variable for a single test assertion to be bad practice. You could make the flag local with an AtomicBoolean. Also please don't swallow the exception, but let the test method rethrow it. Something like this:

```
    @Test
    public void testResourceStoreAccess() throws IOException {
        final AtomicBoolean configurationFileRequested = new AtomicBoolean(false);

        URL url = PreGeneralizedDataStoreFactoryTest.class.getResource("/");
        GeoServerResourceLoader resourceLoader = new GeoServerResourceLoader(URLs.urlToFile(url)) {
            @Override
            public Resource get(String path) {
                if ("geninfo1.xml".equals(path)) {
                    configurationFileRequested.set(true);
                }
                return super.get(path);
            }
        };
 ....
        assertTrue(configurationFileRequested.get());
```